src/auth/impersonated.ts

```diff
@@ -154,4 +160,28 @@ export class Impersonated extends OAuth2Client {
       }
     }
   }
+
+  /**
+   * Generates an OpenID Connect ID token for a service account.
+   *
+   * @param targetAudience the audience for the fetched ID token.
+   */
+  async fetchIdToken(targetAudience: string): Promise<string> {
+    await this.sourceClient.getAccessToken();
+
+    const name = `projects/-/serviceAccounts/${this.targetPrincipal}`;
+    const u = `${this.endpoint}/v1/${name}:generateIdToken`;
+    const body = {
+      delegates: this.delegates,
+      audience: targetAudience,
+      includeEmail: true,
```

Comment:
just a small comment on this.   The `includeEmail` is an optional parameter for the remote api call thats set specifically set here to true.

Certain services on GCP (and def. other places) look for the `email` field in the OIDC token ([IAP programmaic access as an example](https://cloud.google.com/iap/docs/authentication-howto#obtaining_an_oidc_token_in_all_other_cases)... (trust me, it doens't say the email is required in the docs but IAP def needs it)

---

Coming back to the is PR, i don't know much node so I don't see any feasible programmatic structure here to optionally set if includeEmail is required or not.  For example, in go you can:

https://github.com/googleapis/google-api-go-client/blob/main/impersonate/idtoken.go#L30


 If not structurally feasible in node, IMO, its fine to keep default (similar to what golang does here when accessing compute engine api for its id_token (format=full for gce retruns the email)

https://github.com/googleapis/google-api-go-client/blob/main/idtoken/compute.go#L41

Reply:
> Certain services on GCP (and def. other places) look for the email field in the OIDC token ([IAP programmaic access as an example](https://cloud.google.com/iap/docs/authentication-howto#obtaining_an_oidc_token_in_all_other_cases)... (trust me, it doens't say the email is required in the docs but IAP def needs it)

Yep, it's in a blue section here: ["Note: Ensure that the generated ID token has an 'email' claim. This is a requirement for IAP."](https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_service_account)

> Coming back to the is PR, i don't know much node so I don't see any feasible programmatic structure here to optionally set if includeEmail is required or not.

It's pretty trivial; I'll refactor and default it to `true`.